include/conversation.php

```diff
@@ -470,11 +470,14 @@ function item_fieldlists() {
  * @brief SQL join for contacts that are needed for displaying items
  */
 function item_joins() {
-	return "STRAIGHT_JOIN `contact` ON `contact`.`id` = `item`.`contact-id` AND
-		(NOT `contact`.`blocked` OR `contact`.`pending`)
+	return sprintf("STRAIGHT_JOIN `contact` ON `contact`.`id` = `item`.`contact-id`
+		AND `contact`.`rel` IN (%s, %s)
```

Comment:
Please include a check for "self" - because otherwise you won't see your own postings anymore. (The "self" contact has got an undefined "rel" value)

Reply:
But this is only possible if logged in (`public_contact()` is set), so in retrospect we probably shouldn't add the network page conditions in this generic join that may be used in many other places.

Reply:
```public_contact``` (or ```local_user```) is unrelated to this.

BTW: I just realized that there is some problematic query in mod/network.php as well, since it creates a relation with ```author-id```. It is very important to know the difference between ```author-id``` and ```owner-id``` on the one side and ```contact-id``` on the other side.

```author-id``` and ```owner-id``` do link to contact records with ```uid=0``` where ```contact-id``` will always link to a contact with ```uid=local_user()```. The ```contact-id``` refers to the contact where you can select to block or hide someone from your personal network stream.

When you create a post then the the ```contact-id``` will be your ```self``` user while the ```author-id``` will refer a record with your personal ```url``` but the ```uid=0``` and no ```self``` value set.

Means: The above query must be changed into:
```AND (`contact`.`rel` IN (%s, %s) OR self```

Reply:
Okay, I added the change.